src/test/java/com/facebook/openwifirrm/optimizers/tpc/MeasurementBasedApClientTPCTest.java

```diff
@@ -114,4 +122,56 @@ void test1() throws Exception {
 		// Device E: 1 client with RSSI -45 => set to min txPower
 		assertEquals(TPC.MIN_TX_POWER, txPowerMap.get(deviceE).get(UCentralConstants.BAND_5G));
 	}
+
+	@Test
+	@Order(2)
+	void testVariousBands() throws Exception {
+		final String deviceA = "aaaaaaaaaaaa";
+		final String deviceB = "bbbbbbbbbbbb";
+		final String deviceC = "cccccccccccc";
+		final String deviceD = "dddddddddddd";
+
+		DeviceDataManager deviceDataManager = new DeviceDataManager();
+		deviceDataManager.setTopology(
+			TestUtils.createTopology(
+				TEST_ZONE, deviceA, deviceB, deviceC, deviceD
+			)
+		);
+
+		DataModel dataModel = new DataModel();
+		// 2G only
+		dataModel.latestState.put(
+			deviceA,
+			createState(deviceA, 20 /*txPower*/, 20 /*bandwidth*/, new int[]{2} /*channel*/)
+		);
+		// 5G only
+		dataModel.latestState.put(
+			deviceB,
+			createState(deviceB, 20 /*txPower*/, 20 /*bandwidth*/, new int[]{36} /*channel*/)
+		);
+		// 2G and 5G
+		dataModel.latestState.put(
+			deviceC,
+			createState(deviceC, 20 /*txPower*/, 20 /*bandwidth*/, new int[]{2, 36} /*channel*/)
+		);
+		// No valid bands in 2G or 5G
+		dataModel.latestState.put(
+			deviceD,
+			createState(deviceA, 20 /*txPower*/, 20 /*bandwidth*/, new int[]{25} /*channel*/)
+		);
+
+		TPC optimizer = new MeasurementBasedApClientTPC(dataModel, TEST_ZONE, deviceDataManager);
+		Map<String, Map<String, Integer>> txPowerMap =
+			optimizer.computeTxPowerMap();
+
+		Set<String> expectedBands = new TreeSet<>();
```

Comment:
We want to test more than just which bands are present I think. We want to test that the alg actually runs correctly independently on each band. If that doesn't make sense, let me know I can explain more.

Reply:
"the alg actually runs correctly independently" is tested in `testComputeCorrectTxPower`